day2/q2.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+func main() {
+	var wg sync.WaitGroup
+	wg.Add(200)                   // add 200 students to wait group
+	rating := make(chan int, 200) // buffered channel for 200 students
+	for i := 0; i < 200; i++ {
+		go func() {
+			defer wg.Done()
+			time.Sleep(time.Duration(rand.Intn(200)) * time.Millisecond) // to generate random response
+			student := rand.Intn(10)                                     // rating between 1 and 10
+			rating <- student
+		}()
+	}
+
+	wg.Wait()
+	close(rating) // to avoid deadlock  fatal error: all goroutines are asleep - deadlock! this error was coming
+
+	var total int
+	for r := range rating {
+		total += r
+	}
```

Comment:
didn't understand the use of channel in this solution. I can see it used as simple list here, please correct me if I am wrong

Reply:
yes we can use list here as well. Will change this to list if you say so . I used channels since i thought with waitgroups and go routines we can use buffered channels of size 200 to receive the ratings. Arrays me bhi har index pe store karna padega  so we can use either . Nothing as such ki only channels hi use honge yha

Reply:
yes @rzpsarthak13, your explanation is correct.
My intention to ask this question, that, can we create another qo-routine for consuming messages added in the channel. you can try this in the weekend. as of now you can keep this solution. no need to change